src/test/java/org/broadinstitute/hellbender/IntelInflaterDeflaterIntegrationTest.java

```diff
@@ -0,0 +1,148 @@
+package org.broadinstitute.hellbender;
+
+import com.intel.gkl.compression.IntelDeflater;
+import com.intel.gkl.compression.IntelDeflaterFactory;
+import com.intel.gkl.compression.IntelInflaterFactory;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockGunzipper;
+import htsjdk.samtools.util.zip.DeflaterFactory;
+import htsjdk.samtools.util.zip.InflaterFactory;
+import org.broadinstitute.hellbender.utils.NativeUtils;
+import org.broadinstitute.hellbender.utils.RandomDNA;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.test.SamAssertionUtils;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+/**
+ * Test that it's possible to load libIntelInflater
+ */
+public class IntelInflaterDeflaterIntegrationTest extends CommandLineProgramTest {
+
+    private static final String INPUT_FILE = "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.tiny.md.bam";
+
+    @Override
+    public String getTestedToolName() {
+        return "PrintReads";
+    }
+
+    private boolean isIntelInflaterDeflaterSupported() {
+        return (NativeUtils.runningOnLinux() || NativeUtils.runningOnMac()) && !NativeUtils.runningOnPPCArchitecture();
+    }
+
+    @Test
+    public void testIntelInflaterDeflaterIsAvailable() {
+        if ( ! NativeUtils.runningOnLinux()  && ! NativeUtils.runningOnMac()) {
+            throw new SkipException("IntelInflater and IntelDeflater are not available on this platform");
+        }
+
+        if ( NativeUtils.runningOnPPCArchitecture() ) {
+            throw new SkipException("IntelInflater and IntelDeflater are not available for this architecture");
+        }
+
+        Assert.assertTrue(new IntelDeflater().load(null), "IntelInflater/IntelDeflater shared library was not loaded. " +
+                "This could be due to a configuration error, or your system might not support it.");
```

Comment:
I think you need two separate test cases here: one that tries to load the Intel deflater, and another that tries to load the Intel inflater.